pyomo/contrib/doe/doe.py

```diff
@@ -808,12 +830,19 @@ def run_grid_search(
             design_iter = self.design_vars.variable_names_value.copy()
             # update the controlled value of certain time points for certain design variables
             for i, names in enumerate(design_dimension_names):
-                # if the element is a list, all design variables in this list share the same values
-                if isinstance(names, collections.abc.Sequence):
+                if isinstance(names, str):
+                    # if 'names' is simply a string, copy the new value
+                    design_iter[names] = list(design_set_iter)[i]
```

Comment:
What happens if 2 entries are strings?  I think the `design_set_iter` gets exhausted the first time and then could return an empty list for subsequent names.  If you are going to recast the iterator to a list, you probably want to do that once and then cache / reuse the list.

Reply:
I added examples/reactor_design.py to test this case. It works. I added it to the testing suite.